### PR TITLE
 Added pagination link to headers

### LINKED ISSUE #9: Add link for pagination in header
@AntoineDao we will have a number of small utilities that will be shared between services like this one. Right now they are pretty small but at some point we may want to collect them in a separate shared package like `pollination_utils` not to end up with different variation of the same task.

### PR DESCRIPTION
**Pull request by mostaphaRoudsari**
To make this possible I had to relax typing for SensorGridOut to make all the items JSON friendly. I added the types to format so it doesn't affect the opanapi documentation. I also enforced the type by using regex instead of Enum which could not be encoded to JSON.

Resolves #9

### CODE REVIEW

**Comment by mostaphaRoudsari** [comment]
My rebase game has been so poor the last two days! This should also be squash and merged.

**Comment by AntoineDao** [comment]
lol, have you tried out [gitkraken](https://www.gitkraken.com/)? It's pretty good to detangle git spaghetti if you're not comfortable with the CLI (god forbid you'd be using Github Desktop :stuck_out_tongue: )

File: app/models/schemas.py [hunk 1]
```diff
@@ -22,18 +22,12 @@ class Sensor(BaseModel):
...
-
-
-class SensorGridOutType(Enum):
-    type = 'SensorGrid'
-
-
 class SensorGridIn(BaseModel):
     """A flattened collection of sensors for grid-based studies."""
-    type: SensorGridInType
+    type: str = Schema(
```
Comment (AntoineDao) [inline app/models/schemas.py]: This also works: 
```python
type: Enum('SensorGrid', {'type': 'SensorGrid'})
```

File: app/models/schemas.py [hunk 2]
```diff
@@ -47,18 +41,28 @@ class SensorGridIn(BaseModel):
...
+        format='uuid'
+    )
     name: str = Schema(
         ...,
         regex=r'^[.A-Za-z0-9_-]*$'
     )
     sensor_count: int = Schema(
         ...,
-        description="Number of total sensors."
+        description='Number of total sensors.'
```
Comment (AntoineDao) [inline app/models/schemas.py]: Total number of sensors?

File: app/crud/database.py [hunk 3]
```diff
@@ -16,7 +16,15 @@ def get_sensor_grids_db(page, per_page, user: User):
     return grids
 
 
-def get_sensor_grid_db(gid, user: User):
+def count_sensor_grids_db(user: User):
+    """Return number of sensor grids."""
+    session = get_db_session()
+    count = session.query(SensorGrid).filter_by(user_id=str(user.user_id)).count()
```
Comment (AntoineDao) [inline app/crud/database.py]: If you only count over 1 column it'll be a bit quicker:     
```python
count = session.query(SensorGrid.id).filter_by(user_id=str(user.user_id)).count() 
```

File: app/models/schemas.py (hunk 1 above)
Comment (mostaphaRoudsari) [reply app/models/schemas.py]: Enums are not JSON serializable. That's why I changed it to str + regex

**Comment by mostaphaRoudsari** [comment]
@AntoineDao thank you for the comments. Let me update the conftest file and will send a new PR. The interesting part is that all these tests are already merged to master! and that's why I said git is confused. Downloaded GitKraken. Can't see anything wrong with the branches. The tree graphic looks cool!

**Comment by AntoineDao** [comment]
Oh you've probably rebased and merged meaning your commits have the same content but different hash ids so git thinks they're different. Best bet is probably to run a `git reset --soft` to a commit prior to this PR, the pull pollination master and then commit your changes :smile: good luck!

**Comment by mostaphaRoudsari** [comment]
All should be addressed now! @AntoineDao let me know if you have any other comments. :shipit:

**Comment by mostaphaRoudsari** [comment]
I'm going to close this PR and send a new one to trigger Travis.

**Pull request closed without merging**